src/components/select/multi-select/multi-select.pw.tsx

```diff
@@ -1551,6 +1552,29 @@ test("should not add an empty Pill when filter text does not match option text",
   await expect(pillElement).not.toBeVisible();
 });
 
+test("should not select a disabled option when a filter is typed", async ({
+  mount,
+  page,
+}) => {
+  await mount(<MultiSelectWithDisabledOption />);
+
+  await dropdownButton(page).click();
+  const inputElement = selectInput(page);
+  await inputElement.type("t");
```

Comment:
**note**: not essential to add right now, but it seems [Playwright has deprecated the `locator.type` action in a later version](https://playwright.dev/docs/release-notes#deprecations). So going forward I'd recommend using the `locator.fill` action instead for typing text into input fields.